### PR TITLE
zkwasm testing script

### PR DESCRIPTION
**Pull request by MCJOHN974**
(WIP)
Here I added script for easy run and test zkasm. Currently it need proverjs installed in same folder as wasmtime is. Further I will add automatic mode that downlads all necessary files (or maybe add some scripts from proverjs to this repo) for running zkwasm and proverjs preinstalled will not needed. Also will try to add this script to CI.

### CODE REVIEW

**Review by nagisa — APPROVED** [review]
Looks neat so far!

**Comment by MCJOHN974** [comment]
(WIP)
I tried to add some flags that will be useful in CI and in other usecases, but for now flags --install-zkwasm and --install-zkwasm-permanent don't work properly. Anyway, you still can use script without this flags with proverjs preinstalled, it should work.

File: cranelift/test-zk.sh [hunk 1]
```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+
```
Comment (nagisa) [inline cranelift/test-zk.sh:2]: One thing that I typically suggest doing is to `set -eux` and `set -o pipefail`. These provide a much better error handling and debugging opportunities in a black box environment like the CI.

File: cranelift/test-zk.sh [hunk 2]
```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
```
Comment (nagisa) [inline cranelift/test-zk.sh:1]:
```suggestion
#!/usr/bin/env bash
```

There are systems that don't store bash at `/bin` (e.g. NixOS.)

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -38,6 +38,16 @@ concurrency:
...
 jobs:
   # Check Code style quickly by running `rustfmt` over all code
+  Runzk:
+    name: Test generated zkwasm
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        submodules: true
+    - run: ls
```
Comment (akashin) [inline .github/workflows/main.yml:48]: Looks like can this be removed?

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -38,6 +38,16 @@ concurrency:
 
 jobs:
   # Check Code style quickly by running `rustfmt` over all code
+  Runzk:
+    name: Test generated zkwasm
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        submodules: true
```
Comment (akashin) [inline .github/workflows/main.yml:47]: Technically, we don't really need submodules, as we don't build Rust code in this action.

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -38,6 +38,16 @@ concurrency:
 
 jobs:
   # Check Code style quickly by running `rustfmt` over all code
+  Runzk:
+    name: Test generated zkwasm
```
Comment (akashin) [inline .github/workflows/main.yml:42]: Nit: "zkwasm" -> "zkAsm"

File: .github/workflows/main.yml [hunk 6]
```diff
@@ -38,6 +38,16 @@ concurrency:
 
 jobs:
   # Check Code style quickly by running `rustfmt` over all code
+  Runzk:
```
Comment (akashin) [inline .github/workflows/main.yml:41]: Maybe a bit more specific "test_zkasm"?

File: ci/test-zk.sh [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+        echo "Removing temporary installation of zkevm-rom..."
+        rm -rf /tmp/zkevm-rom
+    fi
+
+    exit $exit_code
+
+else
+    zkasm_file="$BASE_DIR/$1"
+    # Replace all "//" with "/"
+    zkasm_file="${zkasm_file//\/\//\/}"
```
Comment (akashin) [inline ci/test-zk.sh:77]: Why do we need this?

File: ci/test-zk.sh [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+
+
+BASE_DIR="../wasmtime"
+
+case $INSTALL_MODE in
+    "permanent")
+        echo "Cloning zkevm-rom into ../../ directory..."
+        git clone https://github.com/0xPolygonHermez/zkevm-rom/ ../../zkevm-rom > /dev/null 2>&1
+        cd ../../zkevm-rom
+        npm install
```
Comment (akashin) [inline ci/test-zk.sh:44]: Consider using https://blog.npmjs.org/post/171556855892/introducing-npm-ci-for-faster-more-reliable here

**Comment by MCJOHN974** [comment]
Hey, @nagisa , I changed bash script a bit and also added this script to CI. As you can see [job](https://github.com/near/wasmtime/actions/runs/6301925196/job/17108045776) runs succesfully, and, [job](https://github.com/near/wasmtime/actions/runs/6301792507/job/17107623262) fails cause of error in add_memory.zkwasm.

File: ci/test-zk.sh [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+        ;;
+    "temporary")
```
Comment (akashin) [inline ci/test-zk.sh:46]: I don't think we would ever use this mode, both in local and remote context, should we remove it?

**Review by akashin — CHANGES_REQUESTED** [review]
Looks super useful, thanks!

File: ci/test-zk.sh [hunk 10]
```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+
+# Colors
+RED='\033[0;31m'
```
Comment (akashin) [inline ci/test-zk.sh:4]: These are not used anymore

File: .github/workflows/main.yml (hunk 6 above)
Comment (nagisa) [reply .github/workflows/main.yml:41]: The comment above is also wrong now.

**Comment by MCJOHN974** [comment]
Hey, @akashin, @nagisa I made some cleanup and the workflow runs as expected. Also i fixed all comments except [this one](https://github.com/near/wasmtime/pull/7#discussion_r1336136868), but I think it is an enchantment and can be added later

**Review by akashin — APPROVED** [review]
Looks good, let's ship it!

**Comment by MCJOHN974** [comment]
Thanks all for your reviews!

**Pull request merged**